textbook/2_grouping.qmd

```diff
@@ -0,0 +1,328 @@
+
+::: {.panel-tabset}
+
+# Altair
+```{python}
+alt.Chart(simpsons_df).mark_circle().encode(
+  color='Group',
+  x=alt.X('X'),
+  y=alt.Y('Y'),
+)
+```
+
+# ggplot
+
+```{r}
+# Load necessary libraries
+library(ggplot2)
+simpsons_df <- read.csv("data/simpsons_paradox_data_l.csv")
+
+ggplot(simpsons_df, aes(x = X, y = Y, color = Group)) +
+  geom_point()
+```
+
+:::
+
+It is critical that each color be clearly distinct from the others, as too-similar colors can confuse the viewer and make differentiation difficult. Similarly, it is wise to limit the number of distinct categories represented by color: adding too many different colors overwhelms the visualization and makes them harder to distinguish.
+
+Accessibility is another key concern when using color. Colorblindness affects approximately 8% of men and 0.5% of women globally, with the most common variant being deuteranopia, which impairs the ability to distinguish between red and green hues. While Altair’s default color palette is designed to be color-blind friendly, ggplot’s is not.
+
+
+### Shape
+
+Shape can be used strategically in combination with color to provide an additional visual cue. This is particularly effective in contexts where colors may be difficult to distinguish. 
+
+A classic example of the power of this combination is the humble traffic light. At first glance, traffic lights seem poorly designed: they use the worst possible colors for people the most common form of colorblindess: red and green. However, in many cities, traffic lights are encoded with using both color and shape. The combination of these encoding channels, alongside the use of predictable y-axis positions for both 'Stop' and 'Go' signals creates a powerful effect that has undoubtedly prevented many accidents.
+
+![](/img/traffic.png)
+
+We can encode shapes aesthetics the same way as color, this can make it easier to distinguish the groups. 
+
+::: {.panel-tabset}
+
+# Altair
+```{python}
+alt.Chart(simpsons_df).mark_circle().encode(
+  shape='Group',
+  color='Group',
+  x=alt.X('X'),
+  y=alt.Y('Y'),
+)
+```
+
+# ggplot
+
+```{r}
+# Load necessary libraries
+library(ggplot2)
+simpsons_df <- read.csv("data/simpsons_paradox_data_l.csv")
+
+ggplot(simpsons_df, aes(x = X, y = Y, shape = Group, color=Group)) +
+  geom_point()
+```
+
+:::
+
+The combination of shape and color can make grouping clearer than either of these categories alone. However, there are some important caveats. At line-drawings or outlines can make color difficult to distinguish as it reduces the area that is colored. It is preferable to use filled shapes when combining these encoding channels. 
+
+Another important consideration is the size of the shapes. It is much easier for the viewer to distinguish between larger shapes than smaller ones. If you have a scatterplot with tiny marks, you may want to restrict the number of unique shapes you show, or consider avoiding shape altogether.
```

Comment:
It would be good if we can give heuristics for the number of shapes. I'm not sure if the rules are the same as color here, my sense is that it might be even lower